commit bab73597ab93bb7477d1a22ebf206e78279232cf
Author: Jonathan Ellis <jbellis@apache.org>
Date:   Fri Sep 23 01:01:23 2011 +0000

    wait a few seconds for index to get flagged as built on startup to avoid false negatives
    
    git-svn-id: https://svn.apache.org/repos/asf/cassandra/branches/cassandra-1.0.0@1174472 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/test/unit/org/apache/cassandra/db/CleanupTest.java b/test/unit/org/apache/cassandra/db/CleanupTest.java
index 7e226a3..db8a035 100644
--- a/test/unit/org/apache/cassandra/db/CleanupTest.java
+++ b/test/unit/org/apache/cassandra/db/CleanupTest.java
@@ -109,7 +109,9 @@ public class CleanupTest extends CleanupHelper
         assertEquals(LOOPS, rows.size());
 
         SecondaryIndex index = cfs.indexManager.getIndexForColumn(COLUMN);
-        assertTrue(index.isIndexBuilt(COLUMN));
+        long start = System.currentTimeMillis();
+        while (!index.isIndexBuilt(COLUMN) && System.currentTimeMillis() < start + 10000)
+            Thread.sleep(10);
 
         // verify we get it back w/ index query too
         IndexExpression expr = new IndexExpression(COLUMN, IndexOperator.EQ, VALUE);
